packages/astro/src/vite-plugin-astro-server/error.ts

```diff
@@ -24,7 +24,10 @@ export function recordServerError(
 	// Our error should already be complete, but let's try to add a bit more through some guesswork
 	const errorWithMetadata = collectErrorMetadata(err, config.root);
 
-	telemetry.record(eventError({ cmd: 'dev', err: errorWithMetadata, isFatal: false }));
+	// Ignore unhandled rejection errors as they appear A LOT and we cannot record the amount to telemetry
+	if (errorWithMetadata.name !== AstroErrorData.UnhandledRejection.name) {
```

Comment:
I don't remember, do we need to be defensive against the error not having a name here?

Reply:
If it doesn't have a name it should follow through to telemetry, shouldn't it? We would only need to be defensive if the error was null or undefined, which seems unlikely.

Reply:
Oh yeah, for sure. I was just wondering if it can be `undefined` at all there or not